### PR TITLE
816 dob via age

### LINKED ISSUE #816: Quest v0.2/ G6PD: Get a patient's DOB by entering their age as an integer
**Name of feature to enhance**
Add a client's DOB by entering their age as an integer 

**Description of feature**
Add on the registration questionnaire for Quest the ability to add age either an integer or actual date, add a selection before the DOB as seen in the screenshot below 

**Describe the enhancement**
1. Add a skip logic question that asks a user to either enter date as a number or date 
2. If Date show a date picker else if number enter a numerical filed
3. On submit, if used date picker submit date, if integer, get the value and compute the DOB i.e if today is **2021**, and entered 21, compute DOB as 2021-21 = **01-01-2020**

NB. This ticket is related to #652 and will be done via a structure map , and adding calculation of attributes in the questionnaire

### PR DESCRIPTION
**Pull request by AbdulWahabMemon**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #816

**Type**
Choose one: Feature

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt [hunk 1]
```diff
@@ -143,6 +145,13 @@ constructor(
 
               if (resource is Patient) resource.managingOrganization = organizationRef
               else if (resource is Group) resource.managingEntity = organizationRef
+              // TODO: calculate birthDate from AgeInput as calculation isn't supported by sdk
+              // https://github.com/google/android-fhir/issues/803
```
Comment (pld) [inline android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt:149]: according to this issue comment, https://github.com/google/android-fhir/issues/803#issuecomment-935834791, all the pre-req issues for this functionality are now closed.

is this now supported?

if so, please remove this and use a calculation.

if not, please add a comment on the issue specify what is remaining so there's a path to closing it and using it here to remove this custom code

Comment (AbdulWahabMemon) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt:149]: **calculated-expression and variable not working - FHIR Reference**
This point is tracked by issue and is not supported
https://github.com/google/android-fhir/wiki/Structured-Data-Capture-Library-Developer's-Guide here calculation is not mentioned in supported list

Comment (pld) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt:149]: Can you please add a comment to https://github.com/google/android-fhir/issues/803 noting that and describing what remains to be implemented?

All the tracking issues in that issue are closed, so it implies that everything's done for calculations to be supported, but like you've said that's not correct, so we need to make sure that is clear and what work remains to be done for this to be supported.

Comment (AbdulWahabMemon) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt:149]: It has been discussed with @maimoonak **calculated-expression and variable ** is yet not supported so as per mutual understanding I have added that //todo comment, and implemented the ticket requirements as per suggestion provided by Maimoona. She has already posted comment to google/android-fhir#803.

Comment (pld) [reply android/engine/src/main/java/org/smartregister/fhircore/engine/ui/questionnaire/QuestionnaireViewModel.kt:149]: Looks good, thanks @AbdulWahabMemon

**Review by pld — APPROVED** [review]

**Comment by pld** [comment]
lgtm! let's bump up the coverage and get it merged

**Review by pld — APPROVED** [review]

**Pull request merged**